community/dbms/src/main/java/org/neo4j/commandline/dbms/ImportCommand.java

```diff
@@ -41,10 +41,15 @@
 import org.neo4j.kernel.impl.util.Validators;
 import org.neo4j.server.configuration.ConfigLoader;
 
+import static org.neo4j.csv.reader.Configuration.DEFAULT;
+import static org.neo4j.unsafe.impl.batchimport.Configuration.DEFAULT_MAX_MEMORY_PERCENT;
+import static org.neo4j.unsafe.impl.batchimport.input.csv.Configuration.COMMAS;
+
 public class ImportCommand implements AdminCommand
 {
     public static final String DEFAULT_REPORT_FILE_NAME = "import.report";
     private static final String[] allowedModes = {"database", "csv"};
+    private static final String[] allowedDelimiterCharacters = {",", "TAB", ";", "|", ":", "#"};
```

Comment:
the list above is the result of a discussion I've had with @praveenag . She thought allowing everything can cause issues if the delimiter interferes with array delimiter or quoting character. My take was to at least have a choice of the most common ones - which is that list.
Any CSV file used by batch importer needs to be specifically built for that purpose due to the conventions being used (think header column == prop name, type encoded in header columns, etc.). Because of that limiting down the allowed delimiters sounds like a good compromise.